Loop/Managers/NightscoutDataManager.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  NightscoutDataManager.swift
+//  Loop
+//
+//  Created by Nate Racklyeft on 8/8/16.
+//  Copyright © 2016 Nathan Racklyeft. All rights reserved.
+//
+
+import Foundation
+import NightscoutUploadKit
+import HealthKit
+//import CarbKit
+import InsulinKit
+import LoopKit
+//import MinimedKit
+
+
+class NightscoutDataManager {
+
+    unowned let deviceDataManager: DeviceDataManager
+
+    init(deviceDataManager: DeviceDataManager) {
+        self.deviceDataManager = deviceDataManager
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(loopDataUpdated(_:)), name: LoopDataManager.LoopDataUpdatedNotification, object: deviceDataManager.loopManager)
+    }
+    
+    @objc func loopDataUpdated(note: NSNotification) {
+        guard
+            let rawContext = note.userInfo?[LoopDataManager.LoopUpdateContextKey] as? LoopDataManager.LoopUpdateContext.RawValue,
+            let context = LoopDataManager.LoopUpdateContext(rawValue: rawContext),
+            case .TempBasal = context
+            else {
+                return
+        }
+        
+        deviceDataManager.loopManager.getLoopStatus { (predictedGlucose, recommendedTempBasal, lastTempBasal, lastLoopCompleted, insulinOnBoard, lastLoopError) in
+            
+            self.deviceDataManager.loopManager.getRecommendedBolus { (bolusUnits, error) in
+                if error != nil {
+                    self.deviceDataManager.logger.addError(error!, fromSource: "LoopManager")
+                }
+                self.uploadLoopStatus(insulinOnBoard, predictedGlucose: predictedGlucose, recommendedTempBasal: recommendedTempBasal, recommendedBolus: bolusUnits, lastTempBasal: lastTempBasal, lastLoopError: lastLoopError)
+            }
+        }
+
+        
+    }
+    
+    private var lastTempBasalUploaded: DoseEntry?
+
+    private func uploadLoopStatus(insulinOnBoard: InsulinValue?, predictedGlucose: [GlucoseValue]?, recommendedTempBasal: LoopDataManager.TempBasalRecommendation?, recommendedBolus: Double?, lastTempBasal: DoseEntry?, lastLoopError: ErrorType?) {
+
+        guard deviceDataManager.remoteDataManager.nightscoutUploader != nil else {
+            return
+        }
+
+        let statusTime = NSDate()
+        
+        // Is this just predictedGlucose[0]?
```

Comment:
Are there situations where `deviceDataManager.glucoseStore?.latestGlucose` will be set, but predictedGlucose?[0] is not?

Looking at LoopDataManager, it fetches `self.deviceDataManager.glucoseStore?.latestGlucose` despite having access to predictedGlucose.